diagnosticsModule/Private/AdfsHealthChecks.ps1

```
@@ -10,25 +10,33 @@ Function TestIsWidRunning()
         {
-            $widService = (Get-WmiObject win32_service | Where-Object {$_.DisplayName.StartsWith("Windows Internal Database")})
-            $widServiceState = $widService.State
-            if ($widServiceState.Count -ne $null -and $widServiceState.Count -gt 1)
+            $service = get-service -name 'MSSQL$MICROSOFT##WID';
+            if($service -ne $null)
             {
-                $widServiceState = $widServiceState[0];
+                $widServiceState=$service.Status;            
+                If ($widServiceState -ne "Running")
+                {
+                    $adfsConfigurationDbTestResult.Result = [ResultType]::Fail;
+                    $adfsConfigurationDbTestResult.Detail = "Current State of WID Service is: $widServiceState";
+                }
+                $widService = (Get-WmiObject win32_service | Where-Object {$_.DisplayName.StartsWith($service.DisplayName)});
+                $adfsConfigurationDbTestResult.Output = @{$serviceStateKey = $widServiceState; $serviceStartModeKey=$widService.StartMode}
+                return $adfsConfigurationDbTestResult;
             }
-            If ($widServiceState -ne "Running")
+            else
             {
-                $adfsConfigurationDbTestResult.Result = [ResultType]::Fail;
-                $adfsConfigurationDbTestResult.Detail = "Current State of WID Service is: $widServiceState";
+                throw "Get-Service with name WIDWriter does not exist.";
             }
-            $adfsConfigurationDbTestResult.Output = @{$serviceStateKey = $widServiceState; $serviceStartModeKey = $widService.StartMode}
-            return $adfsConfigurationDbTestResult;
-        }
+        }        
     }
-    catch [Exception]
+    catch [Exception] 
     {
-        return Create-ErrorExceptionTestResult $testName $_.Exception
-    }
+        $testResult= New-Object TestResult -ArgumentList($testName);
```

is there any reason that we do a catch all exception and just translate them to a not run, this would potentially hide a problem if the user is in a WID configuration and this test runs into an issue. Can we instead check to see the farm is in a wid configuration then do NotRun if it isn't otherwise we do normal execution.

---

Leaving as is for now.  Will be replaced by other work coming later.